src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryTaskLauncher.java

```diff
@@ -91,12 +103,16 @@ public TaskStatus status(String id) {
 	@Override
 	public int getRunningTaskExecutionCount() {
 
-		ListTasksRequest listTasksRequest = ListTasksRequest.builder().state(TaskState.RUNNING).build();
-		return this.client.tasks().list(listTasksRequest).map(listTasksResponse ->
-			listTasksResponse.getPagination().getTotalResults())
-			.doOnError(logError("Failed to list running tasks"))
-			.doOnSuccess(count -> logger.info(String.format("There are %d running tasks", count)))
-			.block(Duration.ofMillis(this.deploymentProperties.getStatusTimeout()));
+		ListTasksRequest listTasksRequest = ListTasksRequest.builder()
+				.state(TaskState.RUNNING)
+				.organizationId(this.organizationId)
```

Comment:
We're already in a bit of a trouble with forced to do blocking calls. These are now adding additional two blocking calls which is even worse. It'd be good to block only once when we need to do it and chain other needed requests within that single blocking call.

Reply:
The two additional blocking calls for space id and org id, if that's what you mean are called once in the constructor and the ids are cached.  Otherwise, this call is the same as always. Sorry about the formatting issues.

Reply:
Uh-oh, blocking network calls in a constructor. I wonder when those are actually getting called?

Reply:
`getCurrentRunningTasks` is called before every launch request. I could move it out of the constructor and lazily cache the ids when this method is call.  Need to consider thread  safety.

Reply:
I think the issue here is that you're trying to use fully imperative programming with reactive api's which usually don't end up well. You could still make those org/space requests with reactive way with a original method and cache those responses within a reactive chain itself. That would then handle caching and also you would not need to worry about threads.